### PR TITLE
Fix golint issues in plugin/pkg/admission

### PR DESCRIPTION
**Pull request by miguelbernadi**
**What this PR does / why we need it**:
Fix golint violations in plugin/pkg/admission tree, as reported in #68026 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
I've added some comments were guidance would be welcome regarding how to tackle some issues, or if to ignore them in this PR.

They will all be squashed before merging.
**Release note**:
```
NONE
```

### CODE REVIEW

File: plugin/pkg/admission/eventratelimit/apis/eventratelimit/register.go
Comment (miguelbernadi) [inline plugin/pkg/admission/eventratelimit/apis/eventratelimit/register.go]: I tried unexporting `SchemeBuilder` and tests would pass, but other packages in the tree are using the same scheme and some can't be unexported. Therefore I keep it here for consistency.

File: plugin/pkg/admission/limitranger/interfaces.go
Comment (miguelbernadi) [inline plugin/pkg/admission/limitranger/interfaces.go]: golint recommends renaming this interface to `Actions` to prevent stutter. This is the last violation remaining in the package and seems reasonable, but would break the interface for consumers. A quick search of users inside the repo shows only 2 uses:
```
vendor/github.com/kubernetes/repo-infra/kazel/kazel.go
613:	options := []string{"limitranger.LimitRangerActions", "BUILD"}

vendor/github.com/bazelbuild/bazel-gazelle/internal/config/config.go
93:var DefaultValidBuildFileNames = []string{"limitranger.LimitRangerActions", "BUILD"}
```

**Comment by miguelbernadi** [comment]
golint suggests exporting the type `nodePlugin` [definition](https://github.com/kubernetes/kubernetes/pull/69941/commits/c60615f11fc7690bd57438c5d6c61464ffa395cb#diff-cbf05e22a4f412de387669881892d46bR63) as it's returned to users but still private, which may be cumbersome. It has plenty of public methods but all fields are private, so it may not be much of an issue to make it public.

It's the only violation left in the package.

**Comment by miguelbernadi** [comment]
golint recommends renaming the struct `PodSecurityPolicyPlugin` to `Plugin` to avoid stutter. There's few usages of this name inside the repo, but it would affect all consumers.

**Comment by yue9944882** [comment]
>  it may be worth splitting the PR into individual PRs. Willing to do it if it's considered worth it.

one pull is fine. just squash after we are sure its ready to be merged.

File: plugin/pkg/admission/eventratelimit/apis/eventratelimit/register.go
Comment (yue9944882) [inline plugin/pkg/admission/eventratelimit/apis/eventratelimit/register.go]: no need for a second line

**Comment by yue9944882** [comment]
> golint suggest the same for podNodeSelector, podPreset, podTolerations and serviceAccount as for nodePlugin. Once the types are exported there will probably be some stutter complaint.

exporting these will be fine i suppose. renaming them to `Plugin`

File: plugin/pkg/admission/eventratelimit/apis/eventratelimit/register.go
Comment (miguelbernadi) [reply plugin/pkg/admission/eventratelimit/apis/eventratelimit/register.go]: I'll address all instances of it ASAP

Comment (miguelbernadi) [reply plugin/pkg/admission/eventratelimit/apis/eventratelimit/register.go]: All instance are now oneline.

**Comment by miguelbernadi** [comment]
> > golint suggest the same for podNodeSelector, podPreset, podTolerations and serviceAccount as for nodePlugin. Once the types are exported there will probably be some stutter complaint.
> 
> exporting these will be fine i suppose. renaming all them to `Plugin`

I'll add that renaming and consequences in a new commit, probably tonight GMT timezone.

**Comment by miguelbernadi** [comment]
@yue9944882 The renaming of the structures is finished and I removed several packages from the golint blacklist. There are several test failures, so I'm trying to find out where did I break them. Some seem flaky but most are directly related to the code.

File: plugin/pkg/admission/limitranger/interfaces.go
Comment (miguelbernadi) [reply plugin/pkg/admission/limitranger/interfaces.go]: That could work. I haven't addressed this yet in case there are other opinions. I'll tackle that once tests are green again.

**Comment by miguelbernadi** [comment]
I missed verifying the tests for podtolerationrestriction and a test was broken. The fix is squashed in the renaming commit.

**Comment by miguelbernadi** [comment]
Rebased to resolve conflicts with master.

File: plugin/pkg/admission/limitranger/interfaces.go
Comment (miguelbernadi) [reply plugin/pkg/admission/limitranger/interfaces.go]: It's done.

**Comment by miguelbernadi** [comment]
@yue9944882 @derekwaynecarr @deads2k All suggested work is complete, review comments addressed and tests are passing. Anything else to be done or should I squash everything into a single commit?

File: plugin/pkg/admission/podtolerationrestriction/doc.go
Comment (yue9944882) [inline plugin/pkg/admission/podtolerationrestriction/doc.go]: seems that its not required by golint, but it's still a good change tho.

File: hack/.golint_failures
Comment (yue9944882) [inline hack/.golint_failures]: can we remove this as well. this comment also applies for L437-438

Comment (miguelbernadi) [reply hack/.golint_failures]: Golint still shows failures for these in my local machine. There are non-autogenerated files called `defaults.go` with golint violations because they invoke autogenerated function names. Example:
```
plugin/pkg/admission/eventratelimit/apis/eventratelimit/v1alpha1/defaults.go:25:6: don't use underscores in Go names; func SetDefaults_Configuration should be SetDefaultsConfiguration
```
I can try and see if the CI will ignore these violations.

File: plugin/pkg/admission/podtolerationrestriction/doc.go
Comment (miguelbernadi) [reply plugin/pkg/admission/podtolerationrestriction/doc.go]: The comment was for a function I had to document and was a good description for the package, so I moved it and adjusted both to golint styles.

File: hack/.golint_failures
Comment (yue9944882) [reply hack/.golint_failures]: > There are non-autogenerated files called defaults.go with golint violations because they invoke autogenerated function names.

oh. apologises. lets keep it if it's generated code

Comment (miguelbernadi) [reply hack/.golint_failures]: Some of the complaints can be fixed, but the ones on using underscores require changing how code is generated. Failure results:
```
Errors from golint:
plugin/pkg/admission/eventratelimit/apis/eventratelimit/v1alpha1/defaults.go:25:1: exported function SetDefaults_Configuration should have comment or be unexported
plugin/pkg/admission/eventratelimit/apis/eventratelimit/v1alpha1/defaults.go:25:6: don't use underscores in Go names; func SetDefaults_Configuration should be SetDefaultsConfiguration
plugin/pkg/admission/resourcequota/apis/resourcequota/v1alpha1/defaults.go:25:1: exported function SetDefaults_Configuration should have comment or be unexported
plugin/pkg/admission/resourcequota/apis/resourcequota/v1alpha1/defaults.go:25:6: don't use underscores in Go names; func SetDefaults_Configuration should be SetDefaultsConfiguration
plugin/pkg/admission/resourcequota/apis/resourcequota/v1beta1/defaults.go:25:1: exported function SetDefaults_Configuration should have comment or be unexported
plugin/pkg/admission/resourcequota/apis/resourcequota/v1beta1/defaults.go:25:6: don't use underscores in Go names; func SetDefaults_Configuration should be SetDefaultsConfiguration
```

Comment (miguelbernadi) [reply hack/.golint_failures]: I removed the commit that verified this breaks in CI.

**Review by yue9944882 — APPROVED** [review]
/lgtm

thx for fixing these! 🎉

/assign @deads2k 
for approval

**Comment by miguelbernadi** [comment]
@deads2k Any input on the PR?

**Comment by miguelbernadi** [comment]
@deads2k ping

**Comment by miguelbernadi** [comment]
Any news on this front? I'm ready to rebase or complete it so it can be merged.

File: plugin/pkg/admission/noderestriction/admission.go
Comment (mourya007) [inline plugin/pkg/admission/noderestriction/admission.go]: Can we add golint comment  for the admitpod() also?

File: plugin/pkg/admission/noderestriction/admission.go
Comment (mourya007) [inline plugin/pkg/admission/noderestriction/admission.go]: Ditto

File: plugin/pkg/admission/noderestriction/admission.go
Comment (mourya007) [inline plugin/pkg/admission/noderestriction/admission.go]: here too.

File: plugin/pkg/admission/noderestriction/admission.go
Comment (miguelbernadi) [reply plugin/pkg/admission/noderestriction/admission.go]: I can add it, but private functions are usually not required to have one.

**Comment by miguelbernadi** [comment]
I'll get to teh rebase and addressing the new comments later today.

**Comment by miguelbernadi** [comment]
@mourya007 Just finished the rebase. Will be addressing your comments now. Bear in mind those methods are unexported, so the comment is usually not required (and golint does not enforce it either).

File: plugin/pkg/admission/noderestriction/admission.go
Comment (mourya007) [reply plugin/pkg/admission/noderestriction/admission.go]: yeah, right but I guess doing same on the private function under the same file is not a big issue. What do you think?

File: plugin/pkg/admission/noderestriction/admission.go
Comment (mourya007) [inline plugin/pkg/admission/noderestriction/admission.go]: getForbiddenUpdateLabels ?

Comment (miguelbernadi) [reply plugin/pkg/admission/noderestriction/admission.go]: I'll update this comment. Can you confirm the descriptions are correct?

**Comment by dims** [comment]
/assign @liggitt 
/milestone v1.15

/approve

(approve for the hack/ changes)

**Comment by liggitt** [comment]
/approve

/hold
please squash before merge

**Comment by miguelbernadi** [comment]
Commits have been squashed. Thanks to everyone that helped!

File: plugin/pkg/admission/limitranger/interfaces.go
Comment (liggitt) [inline plugin/pkg/admission/limitranger/interfaces.go]: please revert this change

File: plugin/pkg/admission/limitranger/interfaces.go
Comment (liggitt) [reply plugin/pkg/admission/limitranger/interfaces.go]: > This is the last violation remaining in the package and seems reasonable, but would break the interface for consumers

renaming the package also breaks the interface

**Comment by liggitt** [comment]
one comment on the package rename, lgtm otherwise

File: plugin/pkg/admission/limitranger/interfaces.go
Comment (miguelbernadi) [reply plugin/pkg/admission/limitranger/interfaces.go]: The rename was needed to prevent failure of this golint rule:
```
Errors from golint:
plugin/pkg/admission/limitranger/interfaces.go:26:6: type name will be used as limitranger.LimitRangerActions by other packages, and that stutters; consider calling this Actions
```
I'm open to suggestions on fixing it. The original approach keeps the same use in all clients and just renames it internally.

Comment (liggitt) [reply plugin/pkg/admission/limitranger/interfaces.go]: I'm not in favor of changing our package and type names to satisfy that lint recommendation

Comment (miguelbernadi) [reply plugin/pkg/admission/limitranger/interfaces.go]: One option would be to rename the type, like the linter suggests, from LimitRangerActions to Actions. Another option is to keep this one blacklisted and unchecked until a future time so we can all agree on a valid solution. Which would you prefer?

Comment (yue9944882) [reply plugin/pkg/admission/limitranger/interfaces.go]: i'm fine w/ both. it's just the smaller change looks more happy.

Comment (miguelbernadi) [reply plugin/pkg/admission/limitranger/interfaces.go]: I've removed the renaming with its effects on all files and restored the blacklist on the limitranger package. I've kept all other linting fixes so it's a smaller effort once there's an agreement on what to do for this particular issue.

Comment (liggitt) [reply plugin/pkg/admission/limitranger/interfaces.go]: > I've removed the renaming with its effects on all files and restored the blacklist on the limitranger package. I've kept all other linting fixes so it's a smaller effort once there's an agreement on what to do for this particular issue.

sounds good, thanks

**Comment by miguelbernadi** [comment]
What's left to merge this PR? Anything I can help with?

**Comment by yue9944882** [comment]
/hold cancel

as @liggitt's comments are all addressed

**Comment by miguelbernadi** [comment]
Seems we need an `/ok-to-test` for tests to run again. @yue9944882

**Comment by liggitt** [comment]
if it makes the milestone, that's fine, but is not required

**Pull request merged**
